src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,58 @@
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Extensions;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+    private readonly ILogger<AddInsurance> _logger;
+
+    public AddInsurance(DataContext context, ILogger<AddInsurance> logger)
+    {
+        _context = context;
+        _logger = logger;   
+    }
+
+    public async Task Handle(InsuranceData item, HttpRequestData req)
+    {
+
+        if (string.IsNullOrEmpty(item.Pid))
+        {
+            _logger.LogWarning("Error, received pid is empty!");
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+        }
+
+        var sub = req.ReadPidFromJwt(); //secured 
```

Comment:
This comment feels useless here. IF you want to comment why get sub from JWT token, say that it saved there or something and that it is provided by AD B2C. But Secured says literally nothing.